### ISSUE TITLE
Buttons are low contrast and focus not visible

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15325]
Buttons are low contrast and focus not visible

-   **Severity:** Medium
-   **Affected Populations:**
    -   Low-Vision
    -   Motor Impaired
    -   Cognitively Impaired
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Document Panel
    -   Publish and Unpublish

#### Issue description

The focus outline for several buttons within the Publish panel have
insufficient color contrast, below the minimum threshold of 3:1 for
interface components:

-   Date picker month navigation buttons : #c4c4c4 (ash) against #fff
    (white): **1.74:1**.
-   Copy link & view post buttons: #bfe7f3 (light blue) against #fff
    (white): **1.31:1**.
-   AM & PM button when not pressed : #bfe7f3 (light blue) against
    #fff (white): **1.31:1**.
-   AM & PM button when pressed : focus outline cannot be seen

All interactive elements should have a visible and obvious focus
indicator, so that sighted keyboard users have a clear indication of
which element has focus. Sufficient color contrast showing states is
important for users who have low-vision or are color-blind, because
states with a low contrast ratio may be difficult or impossible for such
users to see.

##### Issue Code

```
    /* AM, PM buttons, View Post and Copy Links */


    .components-button.is-default:focus:enabled {


        ...


        border-color: #999;


        box-shadow: inset 0 -1px 0 #999, 0 0 0 2px #bfe7f3;


        text-decoration: none;


    }





    /* date picker month navigation buttons */


    .DayPickerNavigation_button__default:focus, .DayPickerNavigation_button__default:hover {


        border: 1px solid #c4c4c4;


        border-bottom-color: ;


    }
```

#### Remediation Guidance

Darken the `border` or `box-shadow` colors used for focus states to
achieve a minimum contrast ratio of 3:1 relative to the white
backgrounds.

Add transparent outlines to ensure that the focus can still be seen by
Windows High Contrast users  and remove the negative `outline-offset` so that the outline doesn't overlap the border.

The combination of `:focus` with `:enabled` seems unsupported for
elements which do not have a `disabled` state in most browsers, so use
another method to style focus states until browser support has
improved.

##### Recommended Code

```
    /* AM, PM buttons, View Post and Copy Links */


    .components-button.is-default:focus {


        ...


        box-shadow: inset 0px 0px 0px 1px #00435d;


        outline: 2px solid transparent;


    }





    /* date picker month navigation buttons */


    .DayPickerNavigation_button__default:focus, .DayPickerNavigation_button__default:hover {


        border: 1px solid #8E8E8E;


        outline: 3px solid transparent;


    }
```

#### Relevant standards

-   [1.4.1 Use of Color
    (Level A)](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-without-color)
-   [1.4.11 Non-text Contrast (Level
    AA)](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-without-color)
-   [2.4.7 Focus Visible (Level
    AA)](https://www.w3.org/TR/WCAG20/#navigation-mechanisms-focus-visible)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-20 in Tenon's report

**Comment by melchoyce** [comment]
Screenshots from the full report:

![image](https://user-images.githubusercontent.com/2846578/57184349-3ae4a900-6e6f-11e9-82a1-b78884c0cb61.png)
![image](https://user-images.githubusercontent.com/2846578/57184351-41732080-6e6f-11e9-802d-cf945e720020.png)

**Comment by afercia** [comment]
+1 see #15432

**Comment by afercia** [comment]
Looks like the AM/PM buttons don't have a focus style when in "pressed" state:

<img width="299" alt="Screenshot 2019-05-10 at 13 50 51" src="https://user-images.githubusercontent.com/1682452/57525613-7e398e80-732b-11e9-9d8b-3b234f914d6c.png">

See #15544.

**Comment by nicolad** [comment]
Hey, @afercia I fixed that here: https://github.com/WordPress/gutenberg/pull/15582
Quick question: how it's possible to find the exact file to which corresponding styles are linked without searching through all project?
Because when I inspected I saw only `style.css` file, no sourcemaps:
![image](https://user-images.githubusercontent.com/1477453/57570775-6a1a8d80-740e-11e9-918a-c99e311d0b32.png)
